src/legacy/ui/public/vis/editors/default/components/default_editor_agg_params_state.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ParamInstance } from './default_editor_agg_params_helper';
+
+export enum AGG_TYPE_ACTION_KEYS {
+  TOUCHED = 'aggTypeTouched',
+  VALIDITY = 'aggTypeValidity',
+}
+
+export interface AggTypeState {
+  touched: boolean;
+  validity: boolean;
```

Comment:
I know this might be stupid now, but I think we should keep those both words in the same form, so if we use `touched` I think we should also rename that property to `valid` and not using a noun here. I would also prefer if we can use both as an adjective (touched/valid), since imho that makes more sense for describing a state than using nouns (validity) here.

Reply:
Renamed.